source/interact-data/raw-queries.txt

```diff
@@ -0,0 +1,383 @@
+In this guide, you can learn how to use {+django-odm+} to run
+raw queries on your MongoDB database. Raw queries allow you to
+query the database by using MongoDB's aggregation pipeline syntax
+rather than Django methods.
+
+The Django QuerySet API provides a ``QuerySet.raw()`` method, which allows
+you to perform raw SQL queries on relational databases. {+django-odm+}
+does not support the ``raw()`` method. Instead, the ODM provides the
+``QuerySet.raw_aggregate()`` method, which you can use to send instructions
+to the database in pipeline stages.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``Movie`` and ``Theater`` models, which
+represent collections in the ``sample_mflix`` database from the :atlas:`Atlas sample datasets </sample-data>`.
+The model classes have the following definitions:
+
+.. code-block:: python
+
+    from django.db import models
+    from django_mongodb_backend.fields import EmbeddedModelField, ArrayField
+    from django_mongodb_backend.managers import MongoManager
+
+    class Movie(models.Model):
+        title = models.CharField(max_length=200)
+        plot = models.TextField(null=True)
+        runtime = models.IntegerField(default=0)
+        released = models.DateTimeField("release date", null=True)
+        awards = EmbeddedModelField(Award)
+        genres = ArrayField(models.CharField(max_length=100), blank=True)
+        objects = MongoManager()
+
+        class Meta:
+            db_table = "movies"
+
+        def __str__(self):
+            return self.title
+    
+    class Theater(models.Model):
+        theaterId = models.IntegerField(default=0)
+        objects = MongoManager()
+
+        class Meta:
+            db_table = "theaters"
+        
+        def __str__(self):
+            return self.title
+
+To learn how to create a Django application that uses a similar ``Movie``
+model to interact with MongoDB documents, visit the :ref:`django-get-started`
+tutorial.
+
+.. _django-raw-queries-run:
+
+Run Raw Queries
+---------------
+
+To run a raw database query, pass an aggregation pipeline
+to the ``QuerySet.raw_aggregate()`` method. Aggregation pipelines
+contain one or more stages that provide instructions on how to
+process documents. After calling the ``raw_aggregate()`` method,
+{+django-odm+} passes your pipeline to the ``pymongo.collection.Collection.aggregate()``
+method and returns the query results as model objects.
+
+.. tip::
+
+   To learn more about constructing aggregation pipelines, see
+   :manual:`Aggregation Pipeline </core/aggregation-pipeline/>`
+   in the {+mdb-server+} manual.
+
+This section shows how to use the ``raw_aggregate()`` method
+to perform the following tasks:
+
+- :ref:`django-raw-queries-filter-group`
+- :ref:`django-raw-queries-search`
+- :ref:`django-raw-queries-geospatial`
+
+.. _django-raw-queries-filter-group:
+
+Filter and Project Document Fields
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This example runs a raw database query by calling the
```

Comment:
```suggestion
The following example runs a raw database query by calling the
```

Reply:
Since I use "following" in the sentence right before the code example (to introduce the pipeline stages used), I removed it from the first sentence to avoid repetition